app/grandchallenge/evaluation/models.py

```diff
@@ -582,9 +591,49 @@ def has_pending_evaluations(self, *, user):
             .exists()
         )
 
+    @property
+    def submission_period_open(self):
+        now = timezone.now()
+        if self.submissions_open and self.submissions_close:
+            submission_period_open = (
+                now > self.submissions_open and now < self.submissions_close
+            )
+        elif self.submissions_open:
+            submission_period_open = now > self.submissions_open
+        elif self.submissions_close:
+            submission_period_open = now < self.submissions_close
+        else:
+            submission_period_open = True
+        return submission_period_open
```

Comment:
Again I find this hard to follow, can you simplify it? I would suggest something like:

```
now = timezone.now()
upper_bound = self.submissions_close or now + timedelta(days=1)
lower_bound = self.submissions_open or now - timedelta(days=1)
return lower_bound < now < upper_bound
```

And is `submission_period_open` the best name? I don't know if it clashes with `submissions_open`. Maybe we should change `submissions_open` to `submissions_open_at` or this method to `is_in_submission_period`? Thoughts welcome :-)